crates/hotshot/src/traits/networking/libp2p_network.rs

```diff
@@ -931,20 +932,22 @@ impl<K: SignatureKey + 'static> ConnectedNetwork<K> for Libp2pNetwork<K> {
             return Err(NetworkError::NotReady);
         };
 
+        // Get the topic subscribers
+        let topic = topic.to_string();
         let topic_map = self.inner.topic_map.read().await;
-        let topic = topic_map
-            .get_by_left(&recipients)
-            .ok_or_else(|| {
-                self.inner.metrics.num_failed_messages.add(1);
-                NetworkError::Libp2p {
-                    source: Box::new(NetworkNodeHandleError::NoSuchTopic),
-                }
-            })?
-            .clone();
 
-        // gossip doesn't broadcast from itself, so special case
-        if recipients.contains(&self.inner.pk) {
-            // send to self
+        // If the topic existed,
+        if let Some(topic_subscribers) = topic_map.get_by_right(&topic) {
```

Comment:
Can we flip the topic map now so the topic is the key and recipients are value.  Or better yet can we remove it and just saved the topics we are subscribed to handle the short circuit?